### PR TITLE
fix variadic

### PR DESCRIPTION
**Pull request by mck**
### **PR Type**
Bug fix, Tests

___

### **Description**
- Fixed variadic input handling in `connect` method.

- Enhanced edge removal logic for variadic inputs.

- Added comprehensive tests for variadic input scenarios.

- Updated changeset to document the fix.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>graph.ts</strong><dd><code>Refactor and fix variadic input handling in graph logic</code>&nbsp; &nbsp; </dd></summary>
<hr>

packages/graph-engine/src/graph/graph.ts

<li>Refactored <code>connect</code> method to handle variadic inputs robustly.<br> <li> Added logic to manage placeholders and edge shifting for variadic <br>indices.<br> <li> Improved edge removal logic to handle variadic inputs and maintain <br>consistency.<br> <li> Simplified and optimized edge creation and propagation.

</details>

  </td>
  <td><a href="https://github.com/tokens-studio/graph-engine/pull/620/files#diff-8441be8add84ffbb1590236774367b226342ce7177e6de010d84b1caea783f61">+196/-108</a></td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>variadic.test.ts</strong><dd><code>Add tests for variadic input handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/graph-engine/tests/suites/graph/variadic.test.ts

<li>Added tests for variadic input edge cases.<br> <li> Verified behavior for negative indices, appending, and placeholder <br>filling.<br> <li> Tested edge shifting and removal for variadic inputs.

</details>

  </td>
  <td><a href="https://github.com/tokens-studio/graph-engine/pull/620/files#diff-f7658da0ed2641daa72c209ac34abba4a693314dd4316bd15d7be40777b5bd70">+130/-0</a>&nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>fluffy-cups-deny.md</strong><dd><code>Add changeset for variadic input fix</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.changeset/fluffy-cups-deny.md

- Documented the fix for variadic input handling.

</details>

  </td>
  <td><a href="https://github.com/tokens-studio/graph-engine/pull/620/files#diff-c50fad787a95d4f53c22256aa99b5e311973b13cc328e5bca78698bb721321ff">+5/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: packages/graph-engine/src/graph/graph.ts [hunk 1]
```diff
@@ -229,34 +229,119 @@ export class Graph {
...
+				const av = a.annotations[annotatedVariadicIndex] ?? 0;
+				const bv = b.annotations[annotatedVariadicIndex] ?? 0;
+				return av - bv;
+			});
+			// if -1 => append
+			if (variadicIndex === -1) {
+				variadicIndex = existingEdges.length;
+			}
+			// if larger => fill placeholders
+			while (existingEdges.length < variadicIndex) {
```
Comment (SorsOps) [inline packages/graph-engine/src/graph/graph.ts:252]: No, lets not do this . Spamming placeholders is not a good idea here. If the issue is that the variadic is causing a problem on load we should rather handle that in the deserialization phase. This will just complicate the system and likely cause more issues

**Comment by SorsOps** [comment]
I also need a better description of the actual problem we're fixing here, which is not apparent from the code, it mostly seems to just be a lot more verbose and inefficient for some of the operations
